src/macros.jl

```diff
@@ -2161,3 +2161,312 @@ julia> @by df :a begin
+
+function unique_helper(x, args...)
+    exprs, outer_flags = create_args_vector(args...)
+    t = (fun_to_vec(ex; no_dest=true, outer_flags=outer_flags) for ex in exprs)
+    tmp = gensym("tmp")
+    quote            
+        $(tmp) = DataFrames.select($x, $(t...); copycols = false)
+        if isempty($(args))
+            $DataFrames.unique($x)
+        else
+            rowidxs = ((!).(DataFrames.nonunique($tmp)))
+            ($x)[rowidxs, :]                
+        end
+    end
+end
+
+"""
+    @unique(d, args...)
+
+Return the first occurence of unique rows in an `AbstractDataFrame` according 
+to given combinations of values in selected columns or their transformation. 
+`args` can be any column selector or transformation accepted by `select`.
+
+The implementation of `@unique` differs from the `unique` function in DataFrames.jl. 
+When `args` are present, `@unique` relies upon an internal `select` call which produces 
+an intermediate data frame containing columns of `df` specified by `args`. The unique rows
+of `df` are thus determined by this intermediate data frame. This focus on `select` allows 
+for multiple arguments to be passed conveniently in the form of column names or transformations.
+
+### Arguments
+
+* `d` : an AbstractDataFrame
+* `args...` :  keyword-like arguments, of the form `:x` designating
+symbols to specify columns or `f(:x)` specifying their transformations 
+
+### Returns
+
+* `::AbstractDataFrame`
+
+Inputs to `@unique` can come in two formats: a `begin ... end` block, or as a series of
+arguments and keyword-like arguments. For example, the following are
+equivalent:
+
+```julia
+@unique df begin 
+    :x + :y
+end
+```
+
+and 
+
+```
+@unique df :x + :y
+```
+
+`@unique` uses the syntax `@byrow` to wrap transformations in
+the `ByRow` function wrapper from DataFrames, apply a function row-wise,
+similar to broadcasting. For example, the call
+
+```
+@unique(df, @byrow :x + :y)
+```
+
+becomes
+
+```
+unique(df, :x => ByRow((x,y) -> x + y))
+```
+
+`@unique` allows `@byrow` at the beginning of a block of selections 
+(i.e. `@byrow begin... end`). The transformation in the block will operate by row.
+
+### Examples
+
+```jldoctest
+julia> using DataFramesMeta;
+
+julia> df = DataFrame(x = 1:10, y = 10:-1:1);
+
+julia> @unique(df, :x + :y)
+1×2 DataFrame
+ Row │ x      y      
+     │ Int64  Int64  
+─────┼───────────────
+   1 │     1      10   
+
+julia> @unique df begin
+            :x + :y
+        end
+1×2 DataFrame
+ Row │ x      y      
+     │ Int64  Int64  
+─────┼───────────────
+   1 │     1      10   
+```
+"""
+macro unique(x, args...)
+    esc(unique_helper(x, args...))
+end
+
+function runique_helper(x, args...)
+    exprs, outer_flags = create_args_vector(args...; wrap_byrow=true)
+    t = (fun_to_vec(ex; no_dest=true, outer_flags=outer_flags) for ex in exprs)
+    tmp = gensym("tmp")
+    quote                    
+        if isempty($(args))
```

Comment:
Having a separate function where we re-implement `DataFrames.unique` also helps reduce code-duplication.